src/polydb.jl

```diff
@@ -0,0 +1,308 @@
+    next = iterate(cursor.mcursor, state)
+    isnothing(next) && return nothing
+    return Polymake.Polydb.parse_document(first(next)), nothing
+end
+
+Base.iterate(coll::Polymake.Polydb.Collection{T}) where T =
+    return iterate(coll, Cursor{T}(find(coll)))
+
+function Base.iterate(coll::Polymake.Polydb.Collection, state::Polymake.Polydb.Cursor)
+    next = iterate(state, nothing)
+    isnothing(next) && return nothing
+    doc, _ = next
+    return doc, state
+end
+
+# functions for `BSON` iteration
+Base.iterate(cursor::Cursor{Mongoc.BSON}, state::Nothing=nothing) =
+    iterate(cursor.mcursor, state)
+
+Base.iterate(coll::Collection{Mongoc.BSON}) =
+   iterate(coll.mcol)
+
+Base.iterate(coll::Collection{Mongoc.BSON}, state::Mongoc.Cursor) =
+   iterate(coll.mcol, state)
+
+#Info
+
+# returns a list of the fields of a collection
+
+function get_fields(coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
+   info1 = Mongoc.find_one(coll_c, Mongoc.BSON("_id" => "info.2.1"))
+   info2 = Mongoc.find_one(coll_c, Mongoc.BSON("_id" => info1["schema"]))
+   schema = info2["schema"]
+   if haskey(schema, "required")
+      return Array{String, 1}(schema["required"])
+   else
+      return _read_fields(schema)
+   end
+end
+
+function _read_fields(a::Array)
+   res = Array{String, 1}()
+   for entry in a
+      append!(res, _read_fields(entry))
+   end
+   return res
+end
+
+function _read_fields(d::Dict)
+   if haskey(d, "required")
+      return d["required"]
+   elseif haskey(d, "then")
+      return _read_fields(d["then"])
+   else
+      return _read_fields(d["allOf"])
+   end
+end
+
+function _print_fields(a::Array{Pair{String, String}})
+   res = ""
+   for (key, value) in a
+      if value == ""
+         res = string(res, key, "\n")
+      else
+         res = string(res, key, ": ", value, "\n")
+      end
+   end
+   return res
+end
+
+# prints information about a specific Collection
+# also used for the info(::Database) function
+function _info(io::IO, coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
+   info = iterate(coll_c)[1]
+   print(io, typeof(coll), "\n", _get_collection(info))
+end
+
+Base.show(io::IO, coll::Collection) = _info(io, coll)
+
+Base.show(io::IO, ::MIME"text/plain", coll::Collection) = print(io, typeof(coll), ": ", coll.mcol.name)
+
+# returns an array containing the names of all collections in the Polydb, also including meta collections
+function _get_collection_names(db::Database)
+   opts = Mongoc.BSON("authorizedCollections" => true, "nameOnly" => true)
+   return Mongoc.get_collection_names(db.mdb;options=opts)
+end
+
+# returns an array cotaining the names of all collections in the Polydb, excluding meta collections
+function get_collection_names(db::Database)
+   names = _get_collection_names(db)
+   res = Array{String, 1}()
+   sizehint!(res, Int64(floor(length(names)/2)))
```

Comment:
```suggestion
   sizehint!(res, floor(Int, length(names)/2))
```
or just `sizehint!(res, div(length(names), 2))`